tests/riscv64/smoke.sh

```diff
@@ -21,55 +21,94 @@ LLVM_BIN_DIR="${LLVM_BIN_DIR}"
 # llvm-cpu bytecode codegen.
 export RISCV_TOOLCHAIN_ROOT="${RISCV_RV64_LINUX_TOOLCHAIN_ROOT}"
 
+# Generate llvm-cpu vmfb with different targets.
+# E.g., mhlo-rv64 | mhlo-rvv | mhlo-x86_64 | tosa-rv64 | tosa-rvv | tosa-x86_64.
 function generate_llvm_cpu_vmfb {
   local target="${1}"; shift
   local compile_args=(
     --iree-hal-target-backends=llvm-cpu
     --iree-llvm-embedded-linker-path="${IREE_HOST_BINARY_ROOT}/bin/lld"
-    --iree-llvm-target-triple=riscv64
-    --iree-llvm-target-cpu=generic-rv64
-    --iree-llvm-target-abi=lp64d
   )
-  if [[ "${target}" == "mhlo" ]]; then
+  if [[ "${target}" == "mhlo"* ]]; then
     compile_args+=(
       --iree-input-type=mhlo
-      --iree-llvm-target-cpu-features="+m,+a,+f,+d,+c"
     )
-  elif [[ "${target}" == "tosa" ]]; then
+  elif [[ "${target}" == "tosa"* ]]; then
     local input_file="${1}"; shift
-    "${IREE_IMPORT_TFLITE_BIN}" -o "${BUILD_RISCV_DIR}/tosa.mlir" "${input_file}"
+    base_file=$(basename ${input_file})
+    "${IREE_IMPORT_TFLITE_BIN}" -o "${BUILD_RISCV_DIR}/${base_file}.tosa.mlir" \
+      "${input_file}"
     compile_args+=(
       --iree-input-type=tosa
-      --iree-llvm-target-cpu-features="+m,+a,+f,+d,+c"
-      "${BUILD_RISCV_DIR}/tosa.mlir"
+      "${BUILD_RISCV_DIR}/${base_file}.tosa.mlir"
     )
-  elif [[ "${target}" == "tosa-rvv" ]]; then
-    local input_file="${1}"; shift
-    "${IREE_IMPORT_TFLITE_BIN}" -o "${BUILD_RISCV_DIR}/tosa.mlir" "${input_file}"
+  fi
+
+  if [[ "${target}" == *"rvv" ]]; then  # RVV cpu features.
     compile_args+=(
-      --iree-input-type=tosa
+      --iree-llvm-target-triple=riscv64
+      --iree-llvm-target-cpu=generic-rv64
+      --iree-llvm-target-abi=lp64d
       --iree-llvm-target-cpu-features="+m,+a,+f,+d,+c,+v"
       --riscv-v-fixed-length-vector-lmul-max=8
       --riscv-v-vector-bits-min=512
-      "${BUILD_RISCV_DIR}/tosa.mlir"
+    )
+  elif [[ "${target}" == *"rv64" ]]; then  # Default rv64 cpu features.
+    compile_args+=(
+      --iree-llvm-target-triple=riscv64
+      --iree-llvm-target-cpu=generic-rv64
+      --iree-llvm-target-abi=lp64d
+      --iree-llvm-target-cpu-features="+m,+a,+f,+d,+c"
+    )
+  elif [[ "${target}" == *"x86_64" ]]; then
+    compile_args+=(
+      --iree-llvm-target-triple=x86_64-linux-pc-elf
     )
   fi
   "${IREE_HOST_BINARY_ROOT}/bin/iree-compile" "${compile_args[@]}" "$@"
 }
 
-generate_llvm_cpu_vmfb mhlo \
+generate_llvm_cpu_vmfb mhlo-rv64 \
   "${ROOT_DIR}/tools/test/iree-run-module.mlir" \
   -o "${BUILD_RISCV_DIR}/iree-run-module-llvm_cpu.vmfb"
 
-wget -P "${BUILD_RISCV_DIR}/" "https://storage.googleapis.com/iree-model-artifacts/person_detect.tflite"
+# The following models will be compiled with both riscv and x86_64 targets for
+# result numerical comparison
+E2E_TFLITE_X64_RISCV_MODELS=(
+  efficientnet_lite0_int8_2.tflite
+)
```

Comment:
This `smoke.sh` script is expanding too much in scope. We should be aiming to use the cross platform test suite ([`tests/e2e/`](https://github.com/iree-org/iree/tree/main/tests/e2e)) for as much coverage as possible. A smoketest running platform specific code or printing platform specific error messages can still be useful though.

Reply:
The function here has been moved into `iree_check_test.cmake`